styx-cli/src/test/java/com/spotify/styx/cli/CliMainTest.java

```diff
@@ -205,7 +207,109 @@ public void testWorkflowDeleteInteractiveForce() throws Exception {
   }
 
   @Test
-  public void testClientError() throws Exception {
+  public void testWorkflowEnable() {
+    final String component = "quux";
+    final WorkflowState workflowState = WorkflowState.builder()
+        .enabled(true)
+        .build();
+
+    when(client.updateWorkflowState(any(), any(), eq(workflowState)))
+        .thenReturn(CompletableFuture.completedFuture(workflowState));
+
+    CliMain.run(cliContext, "workflow", "enable", component, "foo", "bar");
+
+    verify(client).updateWorkflowState(component, "foo", workflowState);
+    verify(client).updateWorkflowState(component, "bar", workflowState);
+    verify(cliOutput).printMessage("Workflow foo in component " + component + " enabled.");
+    verify(cliOutput).printMessage("Workflow bar in component " + component + " enabled.");
+  }
+
+  @Test
+  public void shouldFailToEnableWorkflow() {
+    final String component = "quux";
+    final WorkflowState workflowState = WorkflowState.builder()
+        .enabled(true)
+        .build();
+
+    when(client.updateWorkflowState(any(), any(), eq(workflowState)))
+        .thenReturn(
+            CompletableFuture.completedFuture(WorkflowState.builder().enabled(false).build()));
+
+    CliMain.run(cliContext, "workflow", "enable", component, "foo");
+
+    verify(client).updateWorkflowState(component, "foo", workflowState);
+    verify(cliOutput).printMessage("Failed to enable workflow foo in component " + component + ".");
+  }
+
+  @Test
+  public void testWorkflowDisable() {
+    final String component = "quux";
+    final WorkflowState workflowState = WorkflowState.builder()
+        .enabled(false)
+        .build();
+
+    when(client.updateWorkflowState(any(), any(), eq(workflowState)))
+        .thenReturn(CompletableFuture.completedFuture(workflowState));
+
+    CliMain.run(cliContext, "workflow", "disable", component, "foo", "bar");
+
+    verify(client).updateWorkflowState(component, "foo", workflowState);
+    verify(client).updateWorkflowState(component, "bar", workflowState);
+    verify(cliOutput).printMessage("Workflow foo in component " + component + " disabled.");
+    verify(cliOutput).printMessage("Workflow bar in component " + component + " disabled.");
+  }
+
+  @Test
+  public void shouldFailToDisableWorkflow() {
```

Comment:
I had to read this test a few times to understand what it was doing, perhaps name it `shouldFailWhenReturningUnexpectedState`